From 45d279337ee7f4406feb03666862a8a74e796d4b Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 2 Mar 2022 18:20:29 +0100
Subject: [PATCH 371/529] ARM: dts: stm32: Add OPTEE interrupts to stm32mp131

Describes optee IRQ chip that is used to provide some interrupts
generated by OP-TEE. Adds a phandle to the optee for its interrupts
consumers.

Change-Id: I68bdad2a068e8afce8e9a456179f532079d5cadd
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/280695
---
 arch/arm/boot/dts/stm32mp131.dtsi | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/stm32mp131.dtsi
@@ -37,11 +37,13 @@
 	};
 
 	firmware {
-		optee {
+		optee: optee {
 			method = "smc";
 			compatible = "linaro,optee-tz";
 			interrupt-parent = <&intc>;
 			interrupts = <GIC_PPI 15 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
 		};
 
 		scmi: scmi {
